Add doc mention of datetime and leap seconds (#14136)
astropy/astropy: docs/time/index.rst

datetime   :class:`~astropy.time.TimeDeltaDatetime`
=========  ===================================================

> .. warning:: The ``datetime`` format does *not* support instants of time that

Reviewer: The warning is for `TimeDelta` here -- it should instead be for regular `Time` - roughly line 218, maybe after the note about FITS.

Reviewer 2: Also: datetime does not only not support instances that happen right at the leap seconds, it also makes all the related mistakes regarding actually elapsed time between instances.

>              are leap seconds. This is a 

Reviewer 3: Getting finicky about type checking, but I think a leap second is an interval of time which is different from an instant of time. Maybe just `.. does *not* support times that are within a leap second.`

             `limitation of the Python standard library datetime object <https://github.com/python/cpython/issues/67762>`_

>              not a limitation of Astropy's |Time|. If the upstream datetime
>              object is updated to support leap seconds Astropy can follow it,
>              but until then ``t.datetime`` will raise an exception if ``t`` is 
>              a leap second. See 

Reviewer 3: ```suggestion
             not a limitation of Astropy's |Time|. Until the upstream ``datetime``
             library is updated to support leap seconds ``t.datetime`` will raise 
             an exception if ``t`` is within a leap second.
```

>              `astropy issue #14112 <https://github.com/astropy/astropy/issues/14112>`_ 

Reviewer 3: I'm not sure that referencing #14112 is the best idea here, since that issue meanders a bit and starts with an (incorrect) statement that there is bug in astropy. After some discussion the issue gets to a better understanding but overall I think that the summary version here before this "See ..." sentence gets the essence of issue.
